unliftio/src/UnliftIO/Exception.hs

```diff
@@ -286,13 +289,17 @@ pureTryDeep = unsafePerformIO . tryAnyDeep . return
 
 -- | A helper data type for usage with 'catches' and similar functions.
 --
+-- Since version #(TODO), this is an alias for 'ESafe.Handler' from
+-- @safe-exceptions@ package, originally defined in "Control.Monad.Catch"
+-- from the @exceptions@ package.
+--
 -- @since 0.1.0.0
-data Handler m a = forall e . Exception e => Handler (e -> m a)
+type Handler = ESafe.Handler
```

Comment:
Interesting trick, I never considered using a type synonym for this kind of reexport.

Reply:
It's not actually a great idea unfortunately - you have to use `ExplicitNameSpaces` and write `type Handler, pattern Handler` to re-export the data constructor too. Then they're not bundled together. It's an unfortunate compromise. Which is why this is a bit vestigial - the current code refers to `Handler` in type signatures, but that would possibly make Haddock weird to navigate, since this isn't exported. 

I removed it in favor of re-exporting more directly 😞